### PR TITLE
Cluster Certificate Management

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR documents how users can renew their cluster's PKI certificates through Palette.

## Review Changes

💻 [Preview URL](https://deploy-preview-1308--docs-spectrocloud.netlify.app/clusters/cluster-management/certificate-management)

🎫 [Jira Ticket](https://spectrocloud.atlassian.net/browse/DOC-767?search_id=3c8a939a-c4d3-4cb4-a12a-6e9783bf5a57)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1308#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (rahulhazra97) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:42]:
```suggestion
When you upgrade the control plane on a cluster, the PKI certificates are automatically updated. Upgrading a cluster's Kubernetes version, whether a minor patch or a major release, results in renewed PKI certificates. This is the method that requires the least user actions when it comes to renewing PKI certificates. We recommend regularly updating your clusters to stay up-to-date with security fixes and best practices. By keeping your host cluster up-to-date, you prevent the scenario of PKI certificates from reaching their one-year expiration date.
```

To make this more concise.

**Review by rahulhazra97 — APPROVED** [review]
Some minor changes.

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (karl-cardenas-coding) [reply content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:42]: I'm declining this as we want to call out this method has the least actions.

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:48]: As line 48 talks about updating the pack version, here's another good link to add or to replace the cluster profile link: 

For guidelines on updating pack versions, review [Update the Pack Version](/cluster-profiles/task-update-profile#updatethepackversion).

I used it in the maintenance policy doc, which it talks about updating the pack version when your profile uses a disabled or deleted pack.

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:55]:
```suggestion
You can renew the cluster PKI certificates on-demand using the Palette API or the Palette User Interface (UI). When you manually trigger a certificate renewal action, it results in cluster nodes becoming repaved. Palette will scale up the cluster control plane nodes count and deploy new nodes with new PKI certificates auto-generated by kubeadm. Once the new control plane nodes are active and available, Palette will remove the previous control plane nodes. The worker nodes renew once the control plane nodes are updated and available.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:61]:
```suggestion
A manual renewal of cluster PKI certificates results in all cluster nodes becoming repaved.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [reply content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:55]: Is it "becoming repaved" or "being repaved"? Same question for line 61. I'm not sure if it makes any difference either - just wondered.

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:66]:
```suggestion
Review the following sections to learn more about manually renewing the cluster PKI certificates. Choose your preferred update method, using the Palette UI or the API.
```

Comment (ritawatson) [reply content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:66]: Optionally, you could use a colon:

"...preferred update method: Palette UI or the API."

I switched the order to match the tab order.

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:82]:
```suggestion
- A deployed host cluster.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:95]:
```suggestion
3. Select the host cluster to renew its PKI certificates.
```

Comment (ritawatson) [reply content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:95]: If you want to keep "you want to", it's more correct to say the following, but then it sounds kind of formal:

"Select the host cluster for which you want to renew the PKI certificates."

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:102]:
```suggestion
  ![A view of the Palette UI with an arrow pointing to the **View K8s Certificates** button.](/clusters_cluster-management_certificate-management_cluster-details-page.png)
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:105]:
```suggestion
5. Next, select **Renew All** to start the renewal process.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:115]:
```suggestion
Using the following steps, you can validate that the cluster's PKI certificates were renewed.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:125]:
```suggestion
3. Select the host cluster with the renewed PKI certificates.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:147]:
```suggestion
- A deployed host cluster.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:156]:
```suggestion
1. Set your Palette API key as an environment variable. Add your actual API key in place of `REPLACE_ME`.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:165]:
```suggestion
2. Set the project ID as an environment variable. Add your project ID in place of `REPLACE_ME`. You can find the project ID on the Palette landing page. The project ID is displayed in the top right corner of the page.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:214]:
```suggestion
1. Set your Palette API key as an environment variable. Add your actual API key in place of `REPLACE_ME`. 
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:223]:
```suggestion
2. Set the project ID as an environment variable. Add your project ID in place of `REPLACE_ME`.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:233]:
```suggestion
3. Set the cluster ID as an environment variable. Add your cluster's ID in place of `REPLACE_ME`. 
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:328]:
```suggestion
You can configure Palette to only renew the PKI certificates for the control plane nodes. You can achieve this by using the annotation `spectrocloud.com/cert-renew-controlplane-only` and setting the value to `true`. To enable this behavior, you must use `kubectl` and apply the update to a Custom Resource Definition (CRD) created by Palette during the cluster deployment process.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:341]:
```suggestion
- Access to the host cluster's kubeconfig file. Refer to the [Access Cluster with CLI](/clusters/cluster-management/palette-webctl) guide to learn how to use your cluster's kubeconfig file.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:347]:
```suggestion
1. Set your cluster name as an environment variable. Add your cluster's name in place of `REPLACE_ME`. 
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:364]:
```suggestion
2. Use `kubectl` to update the CRD to include the `spectrocloud.com/cert-renew-controlplane-only` annotation.
```

Comment (ritawatson) [reply content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:364]: Removed because D in CRD is "definition".

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:405]:
```suggestion
3. Select the host cluster to renew its PKI certificates.
```

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:257]:
```suggestion
5. Validate the output and confirm the expiration date is one year away.
```

Comment (ritawatson) [reply content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:257]: For consistency.

**Review by ritawatson — APPROVED** [review]
Just a few comments. The sentence to replace REPLACE_ME uses "place" three times, so I offered alternative wording throughout.

LGTM

**Review by ritawatson — APPROVED** [review]
Just a few comments. The sentence to replace REPLACE_ME uses "place" three times, so I offered alternative wording throughout.

LGTM

File: content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/06-cluster-management/02.5-certificate-management.md:48]:
```suggestion
You can upgrade the Kubernetes version of a cluster by updating the Kubernetes layer of the [cluster profile](/cluster-profiles) and applying the cluster profile update to the cluster. For guidelines on updating pack versions, review [Update the Pack Version](/cluster-profiles/task-update-profile#updatethepackversion).
```

**Pull request merged**
